util/testutil/integration/util.go

```diff
@@ -19,10 +21,10 @@ func startCmd(cmd *exec.Cmd, logs map[string]*bytes.Buffer) (func() error, error
 	if logs != nil {
 		b := new(bytes.Buffer)
 		logs["stdout: "+cmd.Path] = b
-		cmd.Stdout = b
+		cmd.Stdout = &lockingWriter{Writer: b}
 		b = new(bytes.Buffer)
 		logs["stderr: "+cmd.Path] = b
-		cmd.Stderr = b
+		cmd.Stderr = &lockingWriter{Writer: b}
```

Comment:
Seems instantiating another mutex

Reply:
Yes, that is ok. We don't need to synchronize stdout and stderr, only concurrent writes to one of them.

Reply:
If write to stdout and write to stderr happen at the same time, the `bytes.Buffer` may face a race?

Reply:
no, they use different buffers